oeps/oep-0042-arch-authentication.rst

```diff
@@ -0,0 +1,254 @@
+
+.. _third_party_auth (edx-platform): https://github.com/edx/edx-platform/tree/master/common/djangoapps/third_party_auth
+.. _python-social-auth library: https://github.com/omab/python-social-auth
+
+OAuth2 and Authorization
+------------------------
+
+Although OAuth2 is an "authorization framework", this OEP primarily deals with `Authentication (AuthN)`_, and not `Authorization (AuthZ)`_. A future OEP is warranted to cover Authorization in Open edX.  Until that time, here are some important authorization related decisions:
+
+* `OEP-4: Application Authorization (Scopes)`_ (Accepted)
+* `OEP-9: User Authorization (Permissions)`_ (Provisional)
+* `Role Based Access Control (RBAC) Library and ADR`_
+* `Courseware: Use bridgekeeper for Permissions and Tracks`_
+
+.. _`OEP-4: Application Authorization (Scopes)`: https://open-edx-proposals.readthedocs.io/en/latest/oep-0004-arch-oauth-scopes.html
+.. _`OEP-9: User Authorization (Permissions)`: https://open-edx-proposals.readthedocs.io/en/latest/oep-0009-bp-permissions.html
+.. _Role Based Access Control (RBAC) Library and ADR: https://github.com/edx/edx-rbac/blob/master/docs/decisions/0001-role-based-access-control.rst
+.. _`Courseware: Use bridgekeeper for Permissions and Tracks`: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/docs/decisions/0003-permissions-via-bridgekeeper.rst
+
+Standardized Libraries and Utilities
+------------------------------------
+
+This section details a variety of authentication related libraries and utilities that Open edX has standardized on. It is important to keep to these standards in order to help keep Open edX more secure.
+
+For any of the following solutions, it is important to avoid creating local alternatives inside an IDA. If a local alternative exists, it should either be deprecated and replaced by these standards, or requires an ADR explaining why the exception is necessary and how the security of Open edX will continue to be ensured.
+
+API Providers: Authentication Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Django REST Framework (DRF)`_ is the standard library used by Open edX to implement REST APIs in Python. Learn more about `Authentication with Django REST Framework (DRF)`_ here.
+
+The following are all DRF Authentication classes.
+
+.. list-table::
+   :widths: 60 40
+   :header-rows: 1
+
+   * - Authentication Class
+     - Status
+   * - `JwtAuthentication (edx-drf-extensions)`_
+     - Supported
+   * - `SessionAuthentication (django-rest-framework)`_
+     - Supported
+   * - `BearerAuthentication (edx-drf-extensions)`_
+     - Deprecated, except mobile
+   * - `BasicAuthentication (django-rest-framework)`_
+     - * Status Unknown
+       * Default for Blockstore, Video Pipeline, and Discovery
+       * Should this just be valid on specific endpoints for specific purposes?
```

Comment:
Do we know why we need BasicAuth in the platform?
We can consider it as an exception that needs an ADR to explain its outlier usage.

Reply:
* I updated with the ADR requirement.
* I think I know of one use case in ecommerce. But, this would require discovery that I don't plan on doing as part of this.
* I removed the line about the existing default usage.  It will be found when and if anyone tries to do clean-up.